### ISSUE TITLE
poll/vote: should "." *NIX hidden files be discovered by default?

### ISSUE THREAD

**Issue by gauteh** [odeke-em/drive#868]
.git directories are ignored, is this intentional?

**Comment by odeke-em** [comment]
Hello again @gauteh, thanks for the question.

Yes in deed, by default files prefixed by "." are ignored. If you'd like to push them, use flag
`--hidden=true` during a push or pull.
```shell
$ drive push --hidden=true myFiles
$ drive pull --hidden=true repos
```

If you'd like, please send a PR documenting `--hidden` or you can add it to the wiki.

Thank you!

**Comment by gauteh** [comment]
Ok, thanks, seems like it is just: `drive push -hidden A`. I have sent a
pull-request. 

Personally, I would prefer the behavior to be the other way around. This
has caused me to loose some .git repositories from some of my projects.
Not the end of the world, but definetely a bit annoying. 

I feel that drive is more advanced than a GUI client, and it would be
more natural to explicitly ignore the files that should be hidden with
.driveignore, rather than have drive try to guess what I want.

Emmanuel T Odeke writes on februar 4, 2017 1:17:

**Comment by odeke-em** [comment]
@gauteh, thanks for the input.
Yes and no:
No because I'd prefer not to surprise
users who are used to commandline tools behaving normally.
For example `ls` is well
advanced and one of the most *NIX utilities, but it still doesn't list hidden files ie `.` prefixed
files unless you pass in a special flag to have them listed.
 
Yes, because some tools like find actually traverse through even hidden files.

So basically, I see your point and we can change this default behavior but only if many people
complain that not recognizing . prefixed files affects them, then we can have motivation to
change something that potentially might be affecting many users, out of the norm.
Thanks.

**Comment by canpolat** [comment]
I agree with @odeke-em about not changing established the defaults unless there is a critical need for it. In addition to that, you may want to reconsider your workflow about storing your repository on Google Drive:

http://stackoverflow.com/questions/31984751/google-drive-can-corrupt-repositories-in-github-desktop

**Comment by gauteh** [comment]
Hi,

I see the points, in particular since this is how it has been working
until now: one potential caveat is that users doing push -> upgrade ->
pull -> will have the local .prefixed directory deleted.

@Canpolat: I do not think this is a real issue with the 'drive' utility
which is manually invoked, and does not try to synchronize things in the
background simultaneously with repository operations. 

On the other hand, data safety should be one of the absolute highest
priorities. This unexpected did cause loss of data in my case.

In *nix: Any tool that does synchronization, moving, copying, removing
also works on .prefixed files. If you copy a directory, you expect
all the .prefixed sub-directories to also be copied. Anything else would
be unthinkable.

Drive is not a tool that is generally used to list data (I'd be fine
with hiding .prefixed files from 'drive ls'), but rather works on
copying, removing, and synchronization. If following the standard *nix
tools: it should also include .prefixed files. 

Not to mention, I think this is the best way in the first place since
.prefixed files (even if they are not in the current directory) are
skipped without warning and potentially lost, this I think should be the
main priority: never do any harm to user data.

Regards, Gaute

**Comment by odeke-em** [comment]
Wow, @gauteh that was a great counter explanation. It makes sense to me and now my mind is swayed to turning on `--hidden` by default. However, I'll hold my horses before making any change, because as I and @canpolat had explained, changing behaviors for users might cause frustrations since they are used to *NIX utilities behaving some type of way.

Therefore, let's make a poll to collect feedback from users, those who'd like to have `--hidden` turned on by default(so changing the behavior) and those who don't. The reason is that no one in
the past had complained nor requested for this yet folks have been using drive for about 3 years+, and I wouldn't want to suddenly break things for them. I'll repurpose this issue as a poll.

Thank you for the explanations @gauteh and @canpolat.

**Comment by odeke-em** [comment]
I've made a poll on Twitter as well https://twitter.com/odeke_et/status/830874839254589440. Please feel free to share and get other drive users to vote on it, voting is open for 7 days.

At the end of 7 days, I'll collect and count votes then we'll decide what to do.
Thanks!

**Comment by gauteh** [comment]
👍 great, thanks for considering! (voted on twitter as well), I think we should take care that a pull does not result in files being deleted locally after an upgrade if this gets implemented.

**Comment by chevillet** [comment]
I think both behaviours are OK. I wasn't sure how it should be ideally but @gauteh's last post is pretty convincing and I now think `--hiden` should be enabled by default.

As always, I really enjoy the open discussions around `drive`!

**Comment by canpolat** [comment]
Mandatory XKCD link:

![https://xkcd.com/1172/](https://imgs.xkcd.com/comics/workflow.png)

If this was a new feature I would not even think about voting for `--hidden` being the default. And the change would not affect my workflow at all. But since drive has an established used base, I'm hesitant about the move. You never know how people uses the tool. And even though it's not designed that way, there are people who uses `drive` for silent sync in the background. If the popular vote is for a change, then I would suggest having a grace period where `drive` issues a warning about the changing behavior for a while (6 months or so) before the change takes place.

**Comment by odeke-em** [comment]
ahaha @canpolat, great analogy. Now that's an interesting suggestion for a slow rollout, cool!
@chevillet thank you for chiming in with your view and vote!
@gauteh awesome, thanks for casting the first vote!

**Comment by gauteh** [comment]
hehe, yeah - I agree, it is a serious change. We should be very careful
to ensure no data can be lost as a consequence of it.

**Comment by odeke-em** [comment]
Hello there y'all. So the results were finalized back in February
<img width="634" alt="screen shot 2017-04-28 at 10 55 29 pm" src="https://cloud.githubusercontent.com/assets/4898263/25553017/d886d8a2-2c65-11e7-9db0-0b27c85db81b.png">


* Twitter
57% of the folks said No
43% of the folks said Yes
